dist/dfu.js

```diff
@@ -382,7 +382,8 @@
                         case OPCODE.VALIDATE_FIRMWARE:
                             log('complete, reset...');
 
-                            controlChar.service.device.addEventListener("gattserverdisconnected", function() {
+                            server.device.addEventListener("gattserverdisconnected", function() {
```

Comment:
Yes, this was the fix. According to the spec the way you had it should work.. I guess this is a bug I need to report. I'm going to double check this now again.